### ISSUE TITLE
Empty test manufacture field leads to failing certificate validity check

### ISSUE THREAD

**Issue by Ein-Tim** [corona-warn-app/cwa-app-android#3850]
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [X] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [X] Bug is not already reported in another issue

## Technical details

- Device name: Xiaomi A1
- Android version: 9
- App version: 2.6.1

## Describe the bug

In some cases the test manufacture field is empty. This then leads to a failing certificate validity check. 
Here's a screenshot, which a Twitter user [shared](https://twitter.com/ProgrammerDude/status/1421434097889591296?s=20) with me.
He took a RAT in Germany and then checked if he could travel to German with this test. CWA shows this then: 
<img src="https://user-images.githubusercontent.com/67682506/127738760-75921d32-5469-4865-b338-5600f31273ba.png" width="300">

Also, the certificate which CWA created is shown as invalid in CovPassCheck...

## Steps to reproduce the issue

1. Take a RAT
2. Request a DCC
3. Check the DCC with the new validity checker for the country "Germany"
4. See that CWA says that the certificate is not valid

## Expected behaviour

The test manufacture field should be filled. 

## Additional context

1. I report this bug based on [feedback I got on Twitter](https://twitter.com/ProgrammerDude/status/1421434097889591296?s=20).
2. I assume that the testing site did not transmit this information, that's why this field is empty. IHMO testing site should be forced to have to transfer the name of the testing manufacture.

**Comment by thomasaugsten** [comment]
Can you please find out the rat Anbieter

**Comment by Ein-Tim** [comment]
It was a testing site from https://15minutentest.de/de/de

**Comment by arianvp** [comment]
The issued certificate also does not validate in the CovPassCheckApp. 

It was not a one-off error. We issued another one at the test location and that one was also not valid.

**Comment by arianvp** [comment]
The code does validate in the VerificaC19 app by the Italian government.

**Comment by arianvp** [comment]
I'll try another DCC test location (Rotes Kreuz) later today to see if this is an issue with RAT certificates in general or an issue with this specific test location.

**Comment by arianvp** [comment]
I went to another test provider in Berlin and that one _did_ issue a valid certificate!  (https://testaja.de/ts02/)


![Screenshot_20210731-212848](https://user-images.githubusercontent.com/628387/127750556-d1aa5b66-e43b-4d27-abe1-09dafdb2de83.png)

The issue thus seems to be specific to 15minutentest.de

Will you contact them to fix their certificates?
Also seems like RKI  maybe shouldn't issue certificates where the test brand is not specified in the first place.

**Comment by arianvp** [comment]
Unfortunate turn of events. Even though corona-warn-app does validate the certificate from  https://testaja.de , CovPassCheck does _not_ validate :(

![PXL_20210731_194013251](https://user-images.githubusercontent.com/628387/127750792-f6bb8dad-828b-49e4-98b4-75b46b89432e.jpg)

**Comment by arianvp** [comment]
In the case of https://testaja.de the product manufacturer is non-empty but the product name is empty.
![Screenshot_20210731-214511](https://user-images.githubusercontent.com/628387/127750892-1b136fb3-e3a2-4495-8354-10ce46d1c2d5.png)

**Comment by thomasaugsten** [comment]
@arianvp  Can you please provide me the qr code from testaia.de

thomas.augsten@sap.com

**Comment by arianvp** [comment]
Mail has been sent.

**Comment by thomasaugsten** [comment]
This is a complete valid test certficiate. Other check apps and also the iOS CovPass Check App will confirm this.
I can reproduce the issue with Android CovPass Check. But please create a ticket in the Android CovPass Check and I will also forward the issue
https://github.com/Digitaler-Impfnachweis/covpass-android/issues

**Comment by Jo-Achim** [comment]
@thomasaugsten 
> 
> This is a complete valid test certficiate. Other check apps and also the iOS CovPass Check App will confirm this.
> I can reproduce the issue with **_Android CovPass Check_**. But please create a ticket in the Android CovPass Check and I will also forward the issue
> https://github.com/Digitaler-Impfnachweis/covpass-android/issues

See PS in: https://github.com/Digitaler-Impfnachweis/covpass-android/issues/59#issuecomment-890958483.

**Comment by arianvp** [comment]
Any update on this? Has the test provider been notified and has the issue been corrected? Or will all the EU scanner apps be adjusted to cope with this scenario?

How to make sure test providers don't issue invalid certificates in the future?

**Comment by thomasaugsten** [comment]
The test provider was informed.

**Comment by vaubaehn** [comment]
TL;DR

Hi @arianvp ,

> How to make sure test providers don't issue invalid certificates in the future?

With the change in the Corona-TestVerordnung taking effect since August 01, 2021, providers of RATs would need to connect to Corona-Warn-App to receive a refund for the free-of-charge citizen tests. And there are exactly 2 ways to do this:
Either you are participating in the Schnelltest-Portal of T-Systems International (TSI) which offers a nice web-based solution for the whole work flow, or you connect to TSI's backend via your own software implementation.
For the whole bunch of information see here: https://github.com/corona-warn-app/cwa-quicktest-onboarding/wiki 
So, in both cases it's necessary to make a contract with TSI.

In case you want to provide your testees with a EU Digital Covid Certificate (DCC) as a Schnelltest-provider using your own software implementation, you would send testees' data to the TSI backend, which forwards its hash to IBM/Ubirch's backend for signing the DCC. After it had been signed, you may pull the signature for the DCC from TSI, bundle it with testees' data again, and then do what ever you want with it 😉 .
See workflow here: https://github.com/corona-warn-app/cwa-quicktest-onboarding/wiki/Anbindung-an-CWA-mit-Verwendung-von-DCCs#anbindung-an-cwa-mit-verwendung-von-dccs

So, in the case of the empty manufacturer field, there is one and only instance who could check for the validity of the certificates' contents:
**TSI**.

But as all DCC contents are sent to the CWA Backend (=TSI servers) _encrypted_, TSI can not do anything, after the test provider went online and is in "production state".

**BUT**
before the provider gets access to the PROD backend, the provider is required to test its implementation in a WRU environment, and finally needs to pass a "Abnahmetest": https://github.com/corona-warn-app/cwa-quicktest-onboarding/wiki/Checkliste-Abnahmetest

So, the only possibility for TSI to check for such flaws described in this issue would be, to TSI also check contents of produced DCCs extensively. Obviously that didn't happen yet/for all providers.

Probably it would be a good idea, to open an issue in https://github.com/corona-warn-app/cwa-quicktest-onboarding.

But don't expect too much. While I personally perceive the collaboration between SAP and community for CWA as smooth, appreciative and highly constructive, I don't see signs from TSI to value community contributions that much currently.
The reason could be, that they're maybe a bit overwhelmed by workload at the moment - at least I got that impression when looking to issues like
https://github.com/corona-warn-app/cwa-quicktest-onboarding/issues/47 - 3rd party providers can't communicate with TSI because their mailbox is full, or
https://github.com/corona-warn-app/cwa-quicktest-onboarding/issues/31 - 3rd party providers are waiting weeks to get connected to the backend **and the issue gets locked when there are too many complaints, even providers try to connect to each other to get some information***, or
https://github.com/corona-warn-app/cwa-documentation/issues/661#issuecomment-891908744 / https://github.com/corona-warn-app/cwa-quicktest-onboarding/issues/24 , where even an issue that **users are able to manipulate the period of validity of their DCCs when 3rd party providers have a flawed implementation** is apparently considered to be unimportant and it's even not worth to communicate about it.

To be honest, I'm a bit disappointed about TSI's current communication.

But maybe @mlenkeit or @thomasaugsten can add some positive information that could cheer me (or us?) up regarding to this subject?

**Comment by Ein-Tim** [comment]
The initial issue was resolved, test centers have to issue valid certificates, but that's out of the influence of the CWA-Team.  Thus closing this issue.
